src/core/lib/security/security_connector/tls/tls_security_connector.cc

```diff
@@ -180,6 +192,17 @@ void TlsChannelSecurityConnector::check_peer(
       grpc_ssl_peer_to_auth_context(&peer, GRPC_TLS_TRANSPORT_SECURITY_TYPE);
   const TlsCredentials* creds =
       static_cast<const TlsCredentials*>(channel_creds());
+  if (creds->options().server_verification_option() ==
```

Comment:
Please also set the default value for `server_verification_option_` in `grpc_tls_credentials_options.h`

Reply:
And make sure hostname verification check is enabled in e2e tests.